### PR TITLE
[release-2.9] Update to Go v1.22

### PR DESCRIPTION
**Pull request by dhaiducek**
ref: https://issues.redhat.com/browse/ACM-14251

### CODE REVIEW

**Review by gparvin — DISMISSED** [review]

File: .github/workflows/kind.yml [hunk 1]
```diff
@@ -86,7 +86,6 @@ jobs:
     - name: Test Coverage and Report Generation
       run: |
         make test-coverage | tee report_unit.json
-        make coverage-verify
```
Comment (mprahl) [inline .github/workflows/kind.yml:89]: Why was this removed?

Comment (dhaiducek) [reply .github/workflows/kind.yml:89]: Incompatibility with Go 1.21

Comment (dhaiducek) [reply .github/workflows/kind.yml:89]: Details: https://github.com/wadey/gocovmerge doesn't have a `go.mod`, so it's pulling the latest of everything, so we can't use Go 1.21 to run it.
